src/query.c

```diff
@@ -1828,6 +1829,53 @@ int QAST_CheckIsValid(QueryAST *q, IndexSpec *spec, RSSearchOptions *opts, Query
   return QueryNode_CheckIsValid(q->root, spec, opts, status);
 }
 
+static int QueryNode_CheckIsValidAsVectorFilter(QueryNode *n, QueryError *status) {
+  bool withChildren = true;
+  int res = REDISMODULE_OK;
+  switch(n->type) {
+    case QN_VECTOR:
+      if (n->vn.vq->type == VECSIM_QT_KNN) {
+        QueryError_SetError(status, QUERY_ESYNTAX, "VECTOR KNN queries are not allowed in a vector filter");
+        res = REDISMODULE_ERR;
+      } else if (n->vn.vq->type == VECSIM_QT_RANGE) {
+        QueryError_SetError(status, QUERY_ESYNTAX, "VECTOR RANGE queries are not allowed in a vector filter");
+        res = REDISMODULE_ERR;
+      }
+      break;
+    case QN_NULL:
+    case QN_MISSING:
+      withChildren = false;
+      break;
+    default:
+      if (n->opts.explicitWeight) {
+        QueryError_SetError(status, QUERY_ESYNTAX, "Weight attribute is not allowed in a vector filter");
+        res = REDISMODULE_ERR;
+      }
+      break;
+  }
+  // Handle children
+  if (withChildren && res == REDISMODULE_OK) {
+    for (size_t ii = 0; ii < QueryNode_NumChildren(n); ++ii) {
+      res = QueryNode_CheckIsValidAsVectorFilter(n->children[ii], status);
+      if (res == REDISMODULE_ERR)
+        break;
+    }
+  }
+  return res;
+}
+
+// Checks whether QAST is valid as vector filter.
+// It can't contain:
+// Vector KNN search =>[KNN num_neighbours @field $vector]
+// Vector RANGE search @<vector_field>:[VECTOR_RANGE (<radius> | $<radius_param>) $<vector_blob_param> $<vector_query_params>]
+// Weight attribute  =>{weight: X}
+int QAST_CheckIsValidAsVectorFilter(QueryAST *q, QueryError *status) {
+  if (!q || !q->root) {
```

Comment:
Is the coverage report faulty or do we actually not enter here at all?
Consider asserting instead

Reply:
I see it is covered in `QAST_CheckIsValid`, so it's probably a faulty report - just validate please

Reply:
`QAST_CheckIsValidAsVectorFilter()` was removed, now all the validations are done in `QAST_CheckIsValid()`